src/Components/WebAssembly/Build/src/targets/Blazor.MonoRuntime.targets

```diff
@@ -140,6 +147,41 @@
         <TargetOutputPath>$(_BaseBlazorRuntimeOutputPath)%(FileName)%(Extension)</TargetOutputPath>
       </_BlazorOutputWithTargetPath>
 
+      <_BlazorWriteSatelliteAssembly Include="@(_BlazorOutputWithTargetPath->WithMetadataValue('BootManifestResourceType', 'satellite'))" />
+    </ItemGroup>
+
+    <!--
+      When building with BuildingProject=false, satellite assemblies do not get resolved (the ones for the current project and the one for
+      referenced project). BuildingProject=false is typically set for referenced projects when building inside VisualStudio.
+      To workaround this, we'll stash metadata during a regular build, and rehydrate from it when BuildingProject=false.
+    -->
+
+    <PropertyGroup>
+      <_BlazorSatelliteAssemblyStashFile>$(IntermediateOutputPath)blazor\blazor.satelliteasm.props</_BlazorSatelliteAssemblyStashFile>
+    </PropertyGroup>
+
+    <BlazorWriteSatelliteAssemblyFile
+      SatelliteAssembly="@(_BlazorWriteSatelliteAssembly)"
+      WriteFile="$(_BlazorSatelliteAssemblyStashFile)"
+      Condition="'$(BuildInsideVisualStudio)' == 'true' AND '$(BuildingProject)' == 'true' AND '@(_BlazorWriteSatelliteAssembly->Count())' != '0'" />
```

Comment:
Are the conditions `'$(BuildInsideVisualStudio)' == 'true'` and `'@(_BlazorWriteSatelliteAssembly->Count())' != '0'` just a way of trying to minimize the risk of negatively impacting other scenarios?

Because it seems like we should really be doing this even for command-line builds, in case someone later does a publish with `--no-build`, right? And the `_BlazorWriteSatelliteAssembly.Count()` == 0 scenario shouldn't be harmful (we'd just write/read an empty list).